### ISSUE TITLE
Video recording not starting/not stopping

### ISSUE THREAD

**Issue by gregest** [ShareX/ShareX#7393]
Very often when I try to start recording a video, it cannot be stopped and the only thing I can do is repeatedly click Abort until it throws an error or close sharex, this is the only thing it throws which doesn't even seem like an actual error message  

> ffmpeg version n6.1.1-20240114 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 13.2.0 (crosstool-NG 1.25.0.232_c175b21)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-w64-mingw32- --arch=x86_64 --target-os=mingw32 --enable-gpl --enable-version3 --disable-debug --disable-w32threads --enable-pthreads --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libharfbuzz --enable-libvorbis --enable-opencl --disable-libpulse --enable-libvmaf --disable-libxcb --disable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-chromaprint --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libkvazaar --enable-libaribcaption --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-librist --enable-libssh --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libvpl --enable-openal --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --enable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --disable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --disable-doc --disable-ffplay --disable-ffprobe --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags= --extra-libs=-lgomp --extra-version=20240114
  libavutil      58. 29.100 / 58. 29.100
  libavcodec     60. 31.102 / 60. 31.102
  libavformat    60. 16.100 / 60. 16.100
  libavdevice    60.  3.100 / 60.  3.100
  libavfilter     9. 12.100 /  9. 12.100
  libswscale      7.  5.100 /  7.  5.100
  libswresample   4. 12.100 /  4. 12.100
  libpostproc    57.  3.100 / 57.  3.100

A while back after the 16.0.1 update I did have a separate ffmpeg error where recording also crashed, however I haven't encountered that one in a while and instead it got replaced by this error

**Comment by therayyanansari** [comment]
Re-install FFmpeg, if the issue persist let me know.

**Comment by gregest** [comment]
turns out the issue to this specific one was that it for some reason switched to gdigrab instead of dshow, dshow *mostly* works unless I encounter the "conversion failed" error

**Comment by sshere** [comment]
This still happens for me, however it's related to the Microphone.

This happens when the device gets disconnected, but my microphone is currently detected (and was working 20 minutes ago).

**Comment by 50l3r** [comment]
i came from here: https://github.com/ShareX/ShareX/issues/7877

Have you found the solution? Maybe the problem is the same.

**Comment by sshere** [comment]
Nope, unfortunately no real solution found, but uninstalling, deleting the files, and reinstalling the new beta version worked.

I don't think it's fully in the ffmpeg itself, but the logs suggests it's related in some way. I didn't dive into the ShareX code to see what potentially could be causing the issue.

**Comment by 50l3r** [comment]
In my case the video is created well in the folder even with audio.

**Comment by sshere** [comment]
> In my case the video is created well in the folder even with audio.

Yes. The file will always create even if ShareX freezes.  If you’re on windows— run the output in Windows Media Player, and you might notice that windows media player will actually lockup and run in the background without closing.
